os/net/ipv6/sicslowpan.c

```diff
@@ -1615,6 +1619,11 @@ output(const linkaddr_t *localdest)
   }
 #endif /* SICSLOWPAN_COMPRESSION >= SICSLOWPAN_COMPRESSION_IPHC */
 
+#if MAC_CONF_WITH_CSMA && LLSEC802154_CONF_ENABLED
+    packetbuf_set_attr(PACKETBUF_ATTR_SECURITY_LEVEL,
```

Comment:
why not do this in the MAC output function? Would about the check for "CSMA" in sicslowpan

Reply:
It is also set in the CSMA layer - but I think this is for getting a more correct expected size of the header length so that the fragmentation gets good input size for the header - but should it not be setting the correct level instead of 0/1?

Reply:
Having conditionals on the MAC just doesn't feel right. If we need info from the MAC here then let's add a function in the MAC API.

Agree we prob wan the level, not just 0/1

Reply:
I totally agree - how is this handled with TSCH? Is security always on, or is it assuming always there security headers and therefore do not create "full" packets if security level happens to be 0? (if allowed). Optimally there would be no dependencies at all here - on any MAC level features except querying for header size - so maybe the framer calls should also be handled via the NETSTACK MAC - so that the MAC layer could add whatever expected config it needs for configuring the framer?

Reply:
Looks like we have a bug in the TSCH case; security is just not accounted for here :s

Reply:
Yep I think that's the most sensible solution here

Reply:
> Having conditionals on the MAC just doesn't feel right. If we need info from the MAC here then let's add a function in the MAC API.

Agreed. I put it there since I found it equally awkward to put it anywhere else, looking at it now it seems worse than before..

> Maybe sicslowpan.c should call the MAC for getting the header size for this packet - and then the MAC that happen to understand the details of its configuration would be able to calculate it together with the framer.

Or should the MAC rather export a function to perform its own initialization of `packetbuf`?

Reply:
> Or should the MAC rather export a function to perform its own initialization of packetbuf?

Would work for me just as well!

Reply:
I will try to do a PR where we add a MAC function for the length - I do think it is the main (possibly the only) thing 6LoWPAN will need to know - befor creating fragments.

Reply:
The way I solved this (also in #506) was to configure an extra framer that "wraps" the actual framer.